tool/release/createnotes/Note.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.vaticle.dependencies.tool.release.createnotes
+
+import com.eclipsesource.json.Json
+import com.eclipsesource.json.JsonArray
+import com.eclipsesource.json.JsonObject
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelBug
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelFeature
+import com.vaticle.dependencies.tool.release.createnotes.Constant.github
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelPrefix
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelRefactor
+
+class Note {
```

Comment:
The Markdown and CommitInfo files have been restructured into a single class Note:

1. It is responsible for storing two important pieces of information about a commit/PR: the title and the goal.
2. It has two factory methods allowing you to create an instance from a Github commit/PR JSON response: `Note.fromGithubCommit` and `Note.fromGithubPR`.
3. It has the knowledge of rendering markdown with `Note.toMarkdown()` and `Note.toMarkdown(List<Note>) `.